docs/articles/documentation/test-api/assertions/index.md

```diff
@@ -0,0 +1,81 @@
+checked: false
+---
+# Assertions
+
+To check if the state of the tested webpage matches the one you expect to see, use *assertions*.
+
+We will provide a comprehensive [set of assertions](assertion-api.md) and all of them based on Behavior Driven Development style (BDD-style).
+The purpose of BDD style is writing tests that reflect the behavior and identify appropriate tests to writing.
+
+This topic contains the following sections.
+
+* [Assertion timeout](#assertion-timeout)
+* [Assertion options](#assertion-options)
+  * [options.timeout](#optionstimeout)
+
+The following example shows how to check the two values for equality:
+
+```text
+    await t.expect(actual).eql(expected, message, options);
+```
+
+Parameter              | Type     | Description
+---------------------- | -------- | -------------------------------------------------------------------------------
+`actual`                 | Any type | The comparison value
+`expected`                 | Any type | Expected value
+`message`&#160;*(optional)* | String   | Assertion message, will be indicated in the report in case of error.
+`options`&#160;*(optional)* | Object   | See [Options](#assertion-options).
+
+Assertions allow you check values of the standard types (String, Number, Object and etc.) and also assertions can be used
+to check state of the webpage's elements.
+In order to check state of any element on the page in assertion you can use properties and methods of
+[DOM Node Snapshot](../selecting-page-elements/selectors.md#dom-node-snapshots)
+or use [Snapshot API Shorthands](../selecting-page-elements/selectors.md#snapshot-api-shorthands).
+
+The following sample fixture demonstrates how to use assertions for these purposes.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Check property of element', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    await t
+        .expect(developerNameInput().value).eql('', 'input is empty') // Assertion with DOM node snapshot property
+        .typeText(developerNameInput, 'Peter')
+        .expect(developerNameInput.value).eql('Peter', 'input has text'); // Assertion with snapshot API shorthands
+});
+```
+
+## Assertion Timeout
+
+When a assertion is called, TestCafe will waits for its successful completion within the *assertion timeout*.
+
+You can specify the assertion timeout in test code by using the [options.timeout](#assertion-options) option.
+To set this timeout when launching tests, pass it to the [runner.run](../../using-testcafe/programming-interface/runner.md#run)
+method if you use API or specify the [assertion-timeout](../../using-testcafe/command-line-interface.md#--assertion-timeout-ms) option
+if you run TestCafe from the command line.
+
+Within the assertion timeout, the assertion is executed over and over again, until it will passed or the timeout exceeds.
```

Comment:
Within the assertion timeout, the assertion is executed over and over again, until it is passed or the timeout exceeds.